packages/di/src/services/dependency-injection/di-container.class.test.ts

```diff
@@ -0,0 +1,118 @@
+    injectPropertyOne: ProviderOne;
+
+    constructor(
+      private providerTwo: ProviderTwo,
+      private providerThree: ProviderThree,
+      @Inject({ identifier: 'ProviderFive' }) private injectParamProvider: ProviderOne,
+    ) {}
+    testMethod() {
+      return `ProviderFour.testMethod ${this.providerTwo.testMethod()} ${this.injectParamProvider.testMethod()}`;
+    }
+  }
+
+  @Injectable()
+  class ProviderFive {
+    testMethod() {
+      return 'ProviderFive.testMethod';
+    }
+  }
+
+  context.sandbox = sinon.createSandbox();
+
+  context.providers = {
+    providerOne: ProviderOne,
+    providerTwo: ProviderTwo,
+    providerThree: ProviderThree,
+    providerFour: ProviderFour,
+    providerFive: ProviderFive,
+  };
+
+  const onInit = context.sandbox.spy((name, instance) => ({ name, instance }));
+  const onDestroy = context.sandbox.spy((name, instance) => ({ name, instance }));
+
+  context.sandbox.stubs = { onInit, onDestroy };
+
+  context.container = new DiContainer({ onInit, onDestroy });
+
+  context.container.set(context.providers.providerOne, context.providers.providerOne.name);
+  context.container.set(context.providers.providerTwo, context.providers.providerTwo.name);
+  context.container.set(context.providers.providerThree, context.providers.providerThree.name);
+  context.container.set(context.providers.providerFour, context.providers.providerFour.name);
+  context.container.set(context.providers.providerFive, context.providers.providerFive.name);
+});
+
+test.after.each((context) => {
+  context.sandbox.restore();
+  context.container.unsetAll();
+});
+
+test('should set providers', (context) => {
+  const providerFour = context.container.get(context.providers.providerFour.name);
+  assert.instance(context.container.get(context.providers.providerOne.name), context.providers.providerOne);
+  assert.instance(context.container.get(context.providers.providerTwo.name), context.providers.providerTwo);
+  assert.instance(context.container.get(context.providers.providerThree.name), context.providers.providerThree);
+  assert.instance(providerFour, context.providers.providerFour);
+  assert.is(providerFour.testMethod(), 'ProviderFour.testMethod ProviderTwo.testMethod ProviderFive.testMethod');
+  assert.is(providerFour.injectPropertyOne.testMethod(), 'ProviderOne.testMethod');
+  assert.is(context.sandbox.stubs.onInit.callCount, 5);
+});
+
+test('should unset providers', (context) => {
+  // Create instances.
+  context.container.get(context.providers.providerOne.name);
+  context.container.get(context.providers.providerTwo.name);
+  context.container.get(context.providers.providerThree.name);
+  context.container.get(context.providers.providerFour.name);
+
+  // Begin unset.
+  context.container.unset(context.providers.providerOne.name);
+  assert.throws(
+    () => context.container.get(context.providers.providerOne.name),
+    (err) => err instanceof ProviderNotFoundException,
+  );
+
```

Comment:
Let's add an assertion here to make sure this expression `context.container.get(context.providers.providerFour.name);` does not throw until unsetAll() is called in the next line.